### PR TITLE
Add Apache SkyWalking Istio mixer adaptor

### PR DESCRIPTION
**Pull request by wu-sheng**
Apache [SkyWalking](https://github.com/apache/incubator-skywalking) is an open source APM solution, especially designed for microservices, cloud native and container-based (Docker, Kubernetes, Mesos) architectures. 

In the old days, our telemetry data are only from native auto instrument agents(Java, .Net, Nodejs) or friendly ecosystems, such as Zipkin.

Now, we are very interested in the service mesh probe/agent, which could deliver 90% of metric but doesn't need language based agents. 

This PR is the first version of SkyWalking Istio probe, which will fit the first release version of Apache SkyWalking 6.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@ZackButcher @mandarjog @ozevren What are the `ci/circleci: codecov` and `ci/circleci: racetest`? Do I need to add some tests for the adaptor? Could you guide me to some documents?

File: mixer/adapter/skywalking/protocol/service-mesh.proto [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (mandarjog) [inline mixer/adapter/skywalking/protocol/service-mesh.proto]: Asf license? Either include the service def from source or have an istio license

**Comment by wu-sheng** [comment]
> Can you make this a grpc adapter instead?
You can host the code at sky walker or put it in contrib repo.

I think GRPC adapter is not enough, more features are going to be added, such as ID exchange for compression mechanism, tracing header(zipkin, skywalking, jaeger formats) capture, commands downstream.

**Comment by mandarjog** [comment]
@wu-sheng grpc adapters provide semantically identical interface to the compiled in adapter. Anything that is possible using compiled in should be possible using grpc adapter as well.

**Comment by ZackButcher** [comment]
Yeah, I don't want to be the first one piloting gRPC adapters. Unless there's clear documentation for integrating, a fleshed out story around deployment, and documentation on istio.io then I don't think we're ready to force integrations to use gRPC.

**Comment by wu-sheng** [comment]
Following guides, use bypass to do telemetry collection.

**Pull request closed without merging**